internal/pkg/scheduler/scheduler_test.go

```diff
@@ -62,9 +61,14 @@ func TestScheduler(t *testing.T) {
 	})
 
 	g.Go(func() error {
-		wait.WithContext(ctx, 500*time.Millisecond)
-		// return some error here to cause exit error group wait
-		return errTest
+		timer := time.NewTimer(500 * time.Millisecond)
```

Comment:
why replace one liner function with all this?

Reply:
`wait..WithContext` was only being used here. so I decided to remove the wait package to simplify the codebase a bit

Reply:
it was useful function, made this implementation cleaner as well, I'd leave as is.

Reply:
turns out `wait.WithContext` is a copy of `sleep.WithContext`, i'll replace it and keep `wait` deleted